src/Features/VisualBasic/Portable/GenerateConstructor/VisualBasicGenerateConstructorService.vb

```diff
@@ -242,6 +242,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.GenerateConstructor
                     End If
                 End If
 
+                If newNode.Kind = SyntaxKind.AsNewClause Then
+                    ' Speculation is not supported for AsNewClauseSyntax nodes (see https://github.com/dotnet/roslyn/issues/15593).
+                    ' Generate an EqualsValueSyntax node with the inner NewExpression of the AsNewClauseSyntax node for speculation.
+                    
+                    ' The SpeculationAnalyzer would do the work of creating the EqualsValueSyntax for us if we passed the AsNewClause,
+                    ' but we need to build it ourself so that we can locate the newTypeName node in the new tree. This way we can
+                    ' GetSymbolInfo for it.
+
+                    Dim asNewClauseNode = DirectCast(newNode, AsNewClauseSyntax)
+                    newNode = SyntaxFactory.EqualsValue(asNewClauseNode.NewExpression)
+                    newTypeName = DirectCast(newNode.GetAnnotatedNodes(s_annotation).Single(), TypeSyntax)
+                End If
+
                 Dim speculativeModel = SpeculationAnalyzer.CreateSpeculativeSemanticModelForNode(oldNode, newNode, document.SemanticModel)
```

Comment:
is this because if compiler auto generate EqualsValueSyntax, it will rewire AsnewClauseSyntax causing speculation semantic mode's root node is attached to new tree where newTypeName can't be found anymore?

in that case, this feels like a bug on compiler? and can you check where speculativeModel.Tree.GetRoot() is pointing to? speculative one? or original tree? if it is speculative one, you can do GetAnnotatedNodes(s_annotation) to find newTypeName?

Reply:
I think if you do speculativeModel.Tree.GetRoot().GetAnnotatedNodes .. that would be more general form and you don't need multiple GetAnnotatedNodes this code has per different cases?